user_unix.go

```diff
@@ -0,0 +1,68 @@
+// +build linux darwin
+
+package lifecycle
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"syscall"
+)
+
+var currentUser = os.Getuid()
+
+// asUser sets HOME and non-root user credentials on command
+func asUser(cmd *exec.Cmd, uid, gid int) (*exec.Cmd, error) {
```

Comment:
We should probably check that `uid, gid int` actually matches `os.Getuid` and `os.Getgid` when non-root.